extensions/warp-ipfs/src/store/conversation.rs

```diff
@@ -128,11 +127,18 @@ impl ConversationDocument {
     }
 
     pub fn conversation_type(&self) -> ConversationType {
-        match self.settings {
-            ConversationSettings::Direct(_) => ConversationType::Direct,
-            ConversationSettings::Group(_) => ConversationType::Group,
+        if self.recipients.len() <= 2 {
+            ConversationType::Direct
+        } else {
+            ConversationType::Group
         }
     }
+    pub fn has_permission(&self, user: &DID, permission: GroupPermission) -> bool {
+        if let Some(value) = self.permissions.get(&user) {
+            return value.contains(&permission);
+        }
+        return false;
```

Comment:
We could do something like the following to be more optimized

```suggestion
        self.permissions.iter().any(|(id, perms)| id == user && perms.contains(&permission)) 
```

Reply:
done